language-server/modules/langserver-core/src/main/java/org/ballerinalang/langserver/codeaction/providers/changetype/ChangeVariableTypeCodeAction.java

```diff
@@ -0,0 +1,146 @@
+        }
+
+        // Derive possible types
+        List<CodeAction> actions = new ArrayList<>();
+        List<TextEdit> importEdits = new ArrayList<>();
+        List<String> types = CodeActionUtil.getPossibleTypes((TypeSymbol) rhsTypeSymbol, importEdits, context);
+        for (String type : types) {
+            List<TextEdit> edits = new ArrayList<>();
+            edits.add(new TextEdit(CommonUtil.toRange(typeNode.get().lineRange()), type));
+            String commandTitle = String.format(CommandConstants.CHANGE_VAR_TYPE_TITLE, variableName.get(), type);
+            actions.add(createQuickFixCodeAction(commandTitle, edits, context.fileUri()));
+        }
+        return actions;
+    }
+
+    private Optional<ExpressionNode> getTypeNode(Node matchedNode, CodeActionContext context) {
+        switch (matchedNode.kind()) {
+            case LOCAL_VAR_DECL:
+                return Optional.of(
+                        ((VariableDeclarationNode) matchedNode).typedBindingPattern().typeDescriptor());
+            case ASSIGNMENT_STATEMENT:
+                Optional<VariableSymbol> optVariableSymbol = getVariableSymbol(context, matchedNode);
+                if (optVariableSymbol.isEmpty()) {
+                    return Optional.empty();
+                }
+                SyntaxTree syntaxTree = context.workspace().syntaxTree(context.filePath()).orElseThrow();
+                NonTerminalNode node = CommonUtil.findNode(optVariableSymbol.get(), syntaxTree);
+                if (node.kind() == SyntaxKind.TYPED_BINDING_PATTERN) {
+                    return Optional.of(((TypedBindingPatternNode) node).typeDescriptor());
+                } else {
+                    return Optional.empty();
+                }
+            default:
+                return Optional.empty();
+        }
+    }
+
+    private Optional<String> getVariableName(Node matchedNode) {
+        switch (matchedNode.kind()) {
+            case LOCAL_VAR_DECL:
+                VariableDeclarationNode variableDeclrNode = (VariableDeclarationNode) matchedNode;
+                BindingPatternNode bindingPatternNode = variableDeclrNode.typedBindingPattern().bindingPattern();
+                if (bindingPatternNode.kind() != SyntaxKind.CAPTURE_BINDING_PATTERN) {
+                    return Optional.empty();
+                }
+                CaptureBindingPatternNode captureBindingPatternNode = (CaptureBindingPatternNode) bindingPatternNode;
+                return Optional.of(captureBindingPatternNode.variableName().text());
+            case ASSIGNMENT_STATEMENT:
+                AssignmentStatementNode assignmentStmtNode = (AssignmentStatementNode) matchedNode;
+                Node varRef = assignmentStmtNode.varRef();
+                if (varRef.kind() == SyntaxKind.SIMPLE_NAME_REFERENCE) {
+                    return Optional.of(((SimpleNameReferenceNode) varRef).name().text());
+                } else if (varRef.kind() == SyntaxKind.QUALIFIED_NAME_REFERENCE) {
```

Comment:
can we convert the else-if to an if?

Reply:
I am -1 to this. `else-if` is more readable and convey the flow-of-decisions. Since perf factor here is minimal it is just matter of styling.